monitoring.tf

```diff
@@ -0,0 +1,196 @@
+        storageType: memory
+        tag: ${local.fluentbit_tag}.*
+  YAML
+}
+
+# Fluentbit filters to log KaaS pods to cloudwatch
+resource "kubectl_manifest" "fluentbit_cluster_filter_pipeline" {
+  yaml_body = <<-YAML
+    apiVersion: fluentbit.fluent.io/v1alpha2
+    kind: ClusterFilter
+    metadata:
+      name: kaas-pipeline
+      labels:
+        fluentbit.fluent.io/enabled: "true"
+    spec:
+      match: ${local.fluentbit_tag}.*
+      filters:
+      - lua:
+          script:
+            key: containerd.lua
+            name: fluent-bit-containerd-config
+          call: containerd
+          timeAsTable: true
+      - kubernetes:
+          kubeURL: https://kubernetes.default.svc:443
+          kubeCAFile: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+          kubeTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+          kubeTagPrefix: ${local.fluentbit_tag}.var.log.containers
+          labels: true
+          annotations: true
+      - nest:
+          operation: lift
+          nestedUnder: kubernetes
+          addPrefix: kubernetes_
+      - modify:
+          rules:
+          - remove: stream
+          - remove: kubernetes_pod_id
+          - remove: kubernetes_host
+          - remove: kubernetes_container_hash
+      - nest:
+          operation: nest
+          wildcard:
+          - kubernetes_*
+          nestUnder: kubernetes
+          removePrefix: kubernetes_
+      - grep:
+          regex: $kubernetes['annotations']['${var.logging_annotation.name}'] ^${var.logging_annotation.value}$
+  YAML
+}
+
+resource "kubectl_manifest" "fluentbit_cluster_output_pipeline" {
+  yaml_body = <<-YAML
+    apiVersion: fluentbit.fluent.io/v1alpha2
+    kind: ClusterOutput
+    metadata:
+      name: kaas-pipeline
+      labels:
+        fluentbit.fluent.io/enabled: "true"
+    spec:
+      customPlugin:
+        config: |
+          Name cloudwatch_logs
+          Match ${local.fluentbit_tag}.*
+          region ${data.aws_region.current.name}
+          log_group_name ${local.fluentbit_cloudwatch_log_group}
+          log_stream_prefix ${local.fluentbit_cloudwatch_log_stream_prefix}
+          log_stream_template $kubernetes['namespace_name'].$kubernetes['pod_name'].$kubernetes['container_name'].$kubernetes['docker_id']
+          auto_create_group On # Has to be set to On: https://github.com/fluent/fluent-bit/issues/8949
+    YAML
+
+  depends_on = [helm_release.fluent_operator]
+}
+
+# CloudWatch log group and permission to allow fluent-bit to write log stream
+resource "aws_cloudwatch_log_group" "fluentbit" {
+  name              = local.fluentbit_cloudwatch_log_group
+  retention_in_days = var.logging_retention_in_days
+}
+
+data "aws_iam_policy_document" "fluentbit" {
+  statement {
+    sid       = ""
+    effect    = "Allow"
+    resources = ["${aws_cloudwatch_log_group.fluentbit.arn}:*"]
+
+    actions = [
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+    ]
```

Comment:
actions = [
        "logs:CreateLogGroup",
        "logs:CreateLogStream",
        "logs:DescribeLogGroups",
        "logs:DescribeLogStreams",
        "logs:PutRetentionPolicy",
      ]
      
      correct permissions

Reply:
I see theses are not needed since the log group is predefined.